src/clj/puppetlabs/trapperkeeper/services/metrics/metrics_core.clj

```diff
@@ -0,0 +1,344 @@
+  (if (:enabled graphite-config)
+    (let [graphite-sender (build-graphite-sender graphite-config domain)
+          graphite-reporter (build-graphite-reporter (:registry registry-context)
+                                                     metrics-allowed
+                                                     graphite-sender)]
+      (.start graphite-reporter (:update-interval-seconds graphite-config) (TimeUnit/SECONDS))
+      (assoc registry-context :graphite-reporter graphite-reporter))
+    registry-context))
+
+(schema/defn get-graphite-config :- (schema/maybe GraphiteReporterConfig)
+  "Merge together the graphite config for the registry with the global graphite config."
+  [config :- MetricsConfig
+   domain :- schema/Keyword]
+  (let [reporter-config (get-in config [:reporters :graphite])
+        registry-config (get-in config [:registries domain :reporters :graphite])
+        merged-config (merge reporter-config registry-config)]
+    ;; the default value for enabled is false
+    (if (nil? merged-config)
+      merged-config
+      (update-in merged-config [:enabled] #(if (nil? %) false %)))))
+
+(schema/defn get-metrics-allowed :- #{schema/Str}
+  "Get the metrics allowed for the registry. Looks at the metrics-allowed registered for the
+  registry in the registry settings atom using the `initialize-registry-settings` function as well
+  as the metrics-allowed listed in the config file under the `:metrics-allowed` key. Merges these
+  lists together and then adds the metrics prefix to them, returning a set of prefixed allowed
+  metrics."
+  [metrics-config :- MetricsConfig
+   registry-settings :- {schema/Any DefaultRegistrySettings}
+   domain :- schema/Keyword]
+  (let [metric-prefix (get-metric-prefix metrics-config domain)
+        default-metrics-allowed (get-in registry-settings [domain :default-metrics-allowed])
+        configured-metrics-allowed (get-in metrics-config [:registries domain :metrics-allowed])
+        metrics-allowed (concat default-metrics-allowed configured-metrics-allowed)]
+    (construct-metric-names metric-prefix metrics-allowed)))
+
+(schema/defn maybe-add-default-to-config :- MetricsConfig
+  "Add a `:default` key with an empty map as the value to the registries config if it is not
+  present."
+  [metrics-config :- MetricsConfig]
+  (update-in metrics-config [:registries :default] #(if (nil? %) {} %)))
+
+(schema/defn initialize-registries-from-config :- {schema/Any RegistryContext}
+  "Read through the config and create a MetricsRegistry (+ JMX reporter if configured) for every
+  registry mentioned in it. Also create the default registry if not mentioned in the config. Should
+  be called from `init` of the metrics-service."
+  [metrics-config :- MetricsConfig]
+  (let [registries-config (:registries metrics-config)]
+    (into {} (map
+              (fn [x] {x (initialize-registry-context (get registries-config x)
+                                                      (name x))})
```

Comment:
Yeah, I agree.  Since this one was in code I had modified from the original code in PE, I removed it in df5c156.  For the others related to keyword / string conversion, I left the code alone - with the assumption that we'll address those when we do TK-394.